elastic/kibana
# Add SavedObjectsClient.bulkResolve

Resolves #111201.

### Testing

In addition to the included automated tests, some manual testing steps:

* Start Kibana
* Add sample data
* View sample dashboard

You should see a single `_bulk_resolve` API call that contains many `lens` objects to populate the dashboard.

### Notes

In the saved objects core code, I added an `internalBulkResolve` module. Then I replaced the existing `SavedObjectsRepository.resolve` method implementation with that, and I also used it to add a new `bulkResolve` method. So now `resolve` and `bulkResolve` use the same implementation, reducing our testing burden and reducing the size of the SOR itself.

See also the commit message from 1e61891:

> In abstracting this out to a separate module, I was able to reuse this instead of the existing resolve code, and I was able to simplify the unit tests.
> Note, there is currently an inconsistent API response between get and bulkGet (an unsupported type for the former returns 404, and the latter returns 400). I opted to return a 400 for bulkResolveInternal as that is the most consistent response, so that means I had to change the integration tests for resolve accordingly.
